### PR TITLE
Catching unique violation exception on mark-as-read

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: https://github.com/department-of-veterans-affairs/caseflow/issues/3030

When a user calls the mark-as-read endpoint on the same document in quick succession, we throw an ActiveRecord::RecordNotUnique error. We can catch this error and ignore it since it just means the other API call created the appropriate DocumentView record.

**Test Plan**
- [ ] On master change the code in DocumentsController#mark_as_read to:       
```
      DocumentView.find_or_create_by(
        document_id: params[:id],
        user_id: current_user.id) do |t|
        sleep 3
        t.update!(first_viewed_at: Time.zone.now)
      end
      render json: {}
```
Delete all DocumentView records, then open a single document in a new tab twice in quick succession. The unique error should be thrown.
- [ ] On this branch, add the same sleep and open a single document in a new tab twice. Ensure the error is not thrown.

### CODE REVIEW

File: app/controllers/document_controller.rb [hunk 1]
```diff
@@ -33,11 +33,18 @@ def pdf
...
   def mark_as_read
-    DocumentView.find_or_create_by(
-      document_id: params[:id],
-      user_id: current_user.id).tap do |t|
-      t.update!(first_viewed_at: Time.zone.now) unless t.first_viewed_at
+    begin
+      DocumentView.find_or_create_by(
+        document_id: params[:id],
+        user_id: current_user.id) do |t|
+        t.update!(first_viewed_at: Time.zone.now)
```
Comment (tejans24) [inline app/controllers/document_controller.rb]: Doesn't this get updated it every time they look at a document? Should this variable name be changed or should we only update the first time it's been changed (as done with the unless earlier)?

Comment (tejans24) [inline app/controllers/document_controller.rb]: Doesn't this get updated it every time they look at a document? Should this variable name be changed or should we only update the first time it's been changed (as done with the unless earlier)?

Comment (mdbenjam) [reply app/controllers/document_controller.rb]: So if you just pass `find_or_create_by` a block it only runs it on create, not find: https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by. We were doing tap before, but it's unnecessary.

Comment (tejans24) [reply app/controllers/document_controller.rb]: cool, that makes sense.

**Pull request merged**